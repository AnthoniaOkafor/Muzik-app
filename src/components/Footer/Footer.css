footer{
    background: #1ABC9C 0% 0% no-repeat padding-box;
    border-radius: 10px;
    opacity: 1;
    margin: 0% 5% 12% 5% ;
    /* change padding to affect how social icons look */
    padding: 3% 2% 5% 4%;
    color: white;
    font-size: 125%;

}li{
    list-style-type: none;
}
.footerLink {
    display: flex;
}
/* This section is for different plaform Icon */
.download{
    font-size: 18px;
    margin: -110px 0px 0px 64%;
    
}
.download > ul > li{
    display: inline;
}
.download > ul {
    margin-top:17px;
}
.download > .listOfLink > li > a > img{
    width: 30px;
    padding-left: 18px;
}
/* This section is for different social media Icon */
.footerLink > span > ul {
    background-color: white;
    position: absolute;
    padding: 10px 15px;
    top: 9%;
    left: 97.1%;
}
.footerLink > span > .listOfLink > li > a > img{
    width: 30px;
    padding: 18px 0px;
    display: block
}
.footerLink {
    position: relative;

}
/* This section is for List of Links e.g blog */
.linkGroup > ul > li > a{
    text-decoration: none;
    color: white;
    line-height: 36px;
    font-size: 18px;
}
/* inserts space before every list */
.linkGroup > .listOfLink{
    margin: 20px 0px 20px;
}
/* inserts space after every list */
.linkGroup {
    margin-right: 21%;
}



