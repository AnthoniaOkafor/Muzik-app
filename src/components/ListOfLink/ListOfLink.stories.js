import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withKnobs, array } from "@storybook/addon-knobs";

import ListOfLink from ".";

storiesOf("ListOfLink", module)
  .addDecorator(
    withInfo({
      title: "The is component renders a list of any item supplied"
    })
  )
  .addDecorator(withKnobs())
  .add("ListOfLink", () => (
    <Router>
      <ListOfLink item={array("create link", ["link1"])} />
    </Router>
  ));
