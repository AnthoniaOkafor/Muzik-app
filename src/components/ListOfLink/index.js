import React from "react";
import { Link } from "react-router-dom";

const ListOfLink = ({item}) => {
    let list, url
     list = item.map(link=> {
            if(typeof(link) === "string"){
                url = link.replace(/\s+/g, '')
                return <li key={link}><Link to={`/${url}`}>{link}</Link></li>
            }else{
                return <li key={link.link}><a href={link.link}>{link.item}</a></li>
            }
            })     
    
    return(
        <ul className="listOfLink">
            {list}
        </ul>
    )
}

export default ListOfLink