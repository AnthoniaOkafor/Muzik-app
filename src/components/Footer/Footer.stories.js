import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs/register';
import Footer from '.';


storiesOf('Footer', module)
	.addDecorator(
		withInfo({
			text: "This is a footer component"
		})
	)
	.add('footer', () => {
        return <Footer />
    })