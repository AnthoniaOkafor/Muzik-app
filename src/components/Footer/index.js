import React from "react"
import Icon from "../Icon/";
import ListOfLink from "../ListOfLink";

import Header from "../Header";
import Text from "../Text"
import "./Footer.css"

const android = "./assets/android.svg"
const apple = "./assets/apple.svg"
export const facebook = "./assets/facebook.svg"
const linkedin = "./assets/linkedin.svg"
const youtube = "./assets/youtube.svg"

export const platform = [{
        item: <Icon src={apple} alt={'apple icon'}/>,
        link: "/lfdjd"
    },{
        item: <Icon src={android} alt={'android icon'}/>,
        link: "#"
    }]
export let socials = [{
        item: <Icon src={facebook} alt={'facebook icon'}/>,
        link: 'https://www.facebook.com'
    },{
        item: <Icon src={linkedin} alt={'linkedin icon'}/>,
        link: 'https://www.linkedin.com'
    },{
        item: <Icon src={youtube} alt={'youtube icon'}/>,
        link: 'https://www.youtube.com'
    }]
export let link =[{
        Categories: ["All events", "Weddings", "Birthdays", "Karoake", "Clubs", "Get togethers"],
        Resources: ["Media resources", "Terms & conditions", "Privacy policy"],
        QuickLinks: ["Blogs", "Company", "Pricing"],
    }]

const Footer = () => { 

    let group = Object.entries(link[0]).map(([key, value]) => {
        return (
            <div key={key} className="linkGroup">
                <Header heading={key} size={3} color={"white"}/>
                {<ListOfLink item={value}/>}
            </div>
        )
    })
    
    return (
        <footer>
            <div className="footerLink">
             {group}
             <span>
             <ListOfLink item={socials}/>
             </span>
             
            </div>
            <div className="download">
                <Text text="Download App" size="extrasmall"/>
                <ListOfLink item={platform}/>
            </div>
        </footer>
    )

}

export default Footer