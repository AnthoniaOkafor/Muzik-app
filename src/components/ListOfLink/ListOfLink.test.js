import React from "react";
import { shallow } from 'enzyme';
import { socials } from "../Footer/";

import ListOfLink from "./"

let icon = socials
let link = ["Blogs", "Company"]

let setUp = (prop ={}) =>{
    return shallow(<ListOfLink item={prop}/>)
}

describe('List Component', () =>{

    it('should match Icon snap shot', () =>{
        const listSnap = setUp(icon)
        expect(listSnap).toMatchSnapshot();
    })
  
})
describe('renders a list of icons', ()=> {
    let listOfIcon
     beforeEach(()=>{ 
        listOfIcon = setUp(icon)
    })
    it('and find the Icon node', ()=>{
        const icon = listOfIcon.find('Icon');

        expect(icon.length).toBe(3)
    })
})
describe('renders a list of link', ()=> {
    let listOfLink
    beforeEach(()=>{ 
        listOfLink = setUp(link)
    })
    it('and find the list of Links node', ()=>{
        const list = listOfLink.find('Link');

        expect(list.length).toBe(2)
    })
})