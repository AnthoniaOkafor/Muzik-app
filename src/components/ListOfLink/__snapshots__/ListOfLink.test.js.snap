// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component should match Icon snap shot 1`] = `ShallowWrapper {}`;
